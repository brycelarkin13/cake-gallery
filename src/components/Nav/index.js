import React from 'react';

function Nav() {

const categories = [
    {
        name: "wedding",
        description:
        "Photos of customer order wedding cakes",
    },
    { name: "birthday",
      description: 
      "Photos of customer order birthday cakes",
    },
];    

function categorySelected(name) {
    console.log(`${name} clicked`)
}

    return(
        <header>
            <nav className='nav'>
                <ul className='flex-row'>
                    <li className='mx-2'>
                        <a data-testid="link" href="/" className="site-title">
                            FEC
                        </a>
                    </li>
                    {categories.map((category) => (
                        <li
                        className='mx-1'
                        key={category.name}
                        >
                            <span onClick={() => categorySelected(category.name)}>
                                {category.name}
                            </span>
                        </li>
                    ))}
                </ul>
            </nav>
        </header>
    );

}

export default Nav;
