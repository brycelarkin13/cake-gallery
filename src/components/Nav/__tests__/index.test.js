import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '../index';

afterEach(cleanup);

describe('Nav component', () => {
    // baseline test
    it('render', () => {
        render(<Nav />);
    });

    // snapshot test
    it('matches shanpshot', () => {
        const { asFragment } = render(<Nav />);
        expect(asFragment()).toMatchSnapshot();
    });
})
