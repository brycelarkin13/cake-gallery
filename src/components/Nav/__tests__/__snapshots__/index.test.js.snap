// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches shanpshot 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="nav"
    >
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            class="site-title"
            href="/"
          >
            FEC
          </a>
        </li>
        <li
          class="mx-1"
        >
          <span>
            wedding
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            birthday
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
